commit fa2428c3200684467cb9bfe0b87c0656c040b812
Author: Leonardo Arena <rnalrd@alpinelinux.org>
Date:   Wed Jun 10 10:31:31 2020 +0000

    Don't chmod. The package takes care of setting the right permissions for directories and files

diff --git a/lib/private/Config.php b/lib/private/Config.php
index cbdbc5b2..1118981b 100644
--- a/lib/private/Config.php
+++ b/lib/private/Config.php
@@ -242,9 +242,6 @@ class Config {
 		touch($this->configFilePath);
 		$filePointer = fopen($this->configFilePath, 'r+');
 
-		// Prevent others not to read the config
-		chmod($this->configFilePath, 0640);
-
 		// File does not exist, this can happen when doing a fresh install
 		if (!is_resource($filePointer)) {
 			throw new HintException(
diff --git a/lib/private/Log/File.php b/lib/private/Log/File.php
index 6be200f6..7db25286 100644
--- a/lib/private/Log/File.php
+++ b/lib/private/Log/File.php
@@ -82,9 +82,6 @@ class File extends LogDetails implements IWriter, IFileBased {
 	public function write(string $app, $message, int $level) {
 		$entry = $this->logDetailsAsJSON($app, $message, $level);
 		$handle = @fopen($this->logFile, 'a');
-		if ($this->logFileMode > 0 && (fileperms($this->logFile) & 0777) != $this->logFileMode) {
-			@chmod($this->logFile, $this->logFileMode);
-		}
 		if ($handle) {
 			fwrite($handle, $entry."\n");
 			fclose($handle);
diff --git a/lib/private/TempManager.php b/lib/private/TempManager.php
index 49d4ee94..b0943843 100644
--- a/lib/private/TempManager.php
+++ b/lib/private/TempManager.php
@@ -96,7 +96,6 @@ class TempManager implements ITempManager {
 			if ($postFix !== '') {
 				$fileNameWithPostfix = $this->buildFileNameWithSuffix($file, $postFix);
 				touch($fileNameWithPostfix);
-				chmod($fileNameWithPostfix, 0600);
 				$this->current[] = $fileNameWithPostfix;
 				return $fileNameWithPostfix;
 			}
diff --git a/lib/private/legacy/OC_Util.php b/lib/private/legacy/OC_Util.php
index 71f6edba..216abdf8 100644
--- a/lib/private/legacy/OC_Util.php
+++ b/lib/private/legacy/OC_Util.php
@@ -1004,7 +1004,6 @@ class OC_Util {
 			. ' cannot be listed by other users.');
 		$perms = substr(decoct(@fileperms($dataDirectory)), -3);
 		if (substr($perms, -1) !== '0') {
-			chmod($dataDirectory, 0770);
 			clearstatcache();
 			$perms = substr(decoct(@fileperms($dataDirectory)), -3);
 			if ($perms[2] !== '0') {
